File: 1750 minimum-length-of-string-after-deleting-similar-ends/solution.java
class Solution {
    public int minimumLength(String s) {
        int l = 0, r = s.length()-1;
        while(l<r) {
            char curr = s.charAt(l);
            if(s.charAt(r)!=curr) return r-l+1;
            while(l<=r && s.charAt(l)==curr) l++;
            while(l<=r && s.charAt(r)==curr) r--;
        }
        return Math.max(0,r-l+1);
    }
}

File: 1750 minimum-length-of-string-after-deleting-similar-ends/solution.cs
public class Solution {
    public int MinimumLength(string s) {
        int l = 0, r = s.Length-1;
        while(l<r) {
            char curr = s[l];
            if(s[r]!=curr) return r-l+1;
            while(l<=r && s[l]==curr) l++;
            while(l<=r && s[r]==curr) r--;
        }
        return Math.Max(0,r-l+1);
    }
}
